src/sentry/api/endpoints/project_metrics_extraction_rules.py

```diff
@@ -0,0 +1,138 @@
+2. Do we need to prevent breaking something in the project config somehow?
+"""
+
+
+@region_silo_endpoint
+class ProjectMetricsExtractionRulesEndpoint(ProjectEndpoint):
+    publish_status = {
+        "DELETE": ApiPublishStatus.EXPERIMENTAL,
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+        "PUT": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.TELEMETRY_EXPERIENCE
+
+    def has_feature(self, organization, request):
+        return features.has(
+            "organizations:custom-metrics-extraction-rule", organization, actor=request.user
+        )
+
+    def delete(self, request: Request, project: Project) -> Response:
+        """DELETE an extraction rule in a project. Returns 204 No Data on success."""
+        if not self.has_feature(project.organization, request):
+            return Response(status=404)
+
+        rules_update = request.data.get("metricsExtractionRules") or ""
+        if len(rules_update) == 0:
+            return Response(status=204)
+
+        try:
+            state_update = self._deserialize_deleted_rules_update(rules_update)
+            delete_metrics_extraction_rules(project, state_update)
+        except Exception as e:
+            return Response(status=500, data={"detail": str(e)})
+
+        return Response(status=204)
+
+    def get(self, request: Request, project: Project) -> Response:
+        """GET extraction rules for project. Returns 200 and a list of extraction rules on success."""
+        if not self.has_feature(project.organization, request):
+            return Response(status=404)
+
+        try:
+            extraction_rules = get_metrics_extraction_rules(project)
+        except Exception as e:
+            return Response(status=500, data={"detail": str(e)})
+
+        return Response(status=200, data=extraction_rules)
+
+    def post(self, request: Request, project: Project) -> Response:
+        """POST an extraction rule to create a resource."""
+        if not self.has_feature(project.organization, request):
+            return Response(status=404)
+
+        rules_update = request.data.get("metricsExtractionRules")
+
+        if not rules_update or len(rules_update) == 0:
+            return Response(
+                status=400,
+                data={"detail": "Please specify the metric extraction rule to be created."},
+            )
+
+        try:
+            state_update = self._deserialize_rules_update(rules_update)
+            persisted_rules = create_metrics_extraction_rules(project, state_update)
+            updated_rules = serialize(
+                persisted_rules, request.user, MetricsExtractionRuleSerializer()
+            )
+        except Exception as e:
+            return Response(status=400, data={"detail": str(e)})
+
+        return Response(status=200, data=updated_rules)
+
+    def put(self, request: Request, project: Project) -> Response:
```

Comment:
@shellmayr As of how the rules are identified via the MRI, this only allows updating the tags, right?
Afaik, as soon as something else is updated a new rule will be created?